Build error: "‘cv’ does not name a type" with USE_OPENCV = 0

While building from source, an error:
```
src/io/image_det_aug_default.cc:18:14: error: ‘cv’ does not name a type
 using Rect = cv::Rect_<float>;
              ^
Makefile:207: recipe for target 'build/src/io/image_det_aug_default.o' failed
make: *** [build/src/io/image_det_aug_default.o] Error 1

```
is thrown.

This is weired because USE_OPENCV = 0 in config.mk.

## Environment info
Operating System:
Ubuntu 16.04
Or if installed from source:

MXNet commit hash (`git rev-parse HEAD`):
911f7474deafc3cc95a2589b89432f94f22cbbe1


## Error Message:
src/io/image_det_aug_default.cc:18:14: error: ‘cv’ does not name a type
 using Rect = cv::Rect_<float>;
              ^
Makefile:207: recipe for target 'build/src/io/image_det_aug_default.o' failed
make: *** [build/src/io/image_det_aug_default.o] Error 1


## Minimum reproducible example
Easily reproducible by following the step by step below on a newly installed ubuntu 16.04 without opencv. 
## Steps to reproduce
1. git clone --recursive https://github.com/dmlc/mxnet.git
2. cd mxnet/
3. emacs make/config.mk -> change USE_OPENCV = 1 to USE_OPENCV = 0
4. make -j8

It compiled fine on the same system with the same configuration with mxnet version 2735414c9f55e3e33d7eeb0c0293e0175e2644d6

--- a/src/io/image_det_aug_default.cc
+++ b/src/io/image_det_aug_default.cc
@@ -183,7 +183,6 @@ std::vector<dmlc::ParamFieldInfo> ListDefaultDetAugParams() {
   return DefaultImageDetAugmentParam::__FIELDS__();
 }
 
-  
 #if MXNET_USE_OPENCV
 using Rect = cv::Rect_<float>;
 

--- a/src/io/image_det_aug_default.cc
+++ b/src/io/image_det_aug_default.cc
@@ -14,6 +14,9 @@
 
 namespace mxnet {
 namespace io {
+
+#if MXNET_USE_OPENCV
+
 using nnvm::Tuple;
 using Rect = cv::Rect_<float>;
 
@@ -183,7 +186,6 @@ std::vector<dmlc::ParamFieldInfo> ListDefaultDetAugParams() {
   return DefaultImageDetAugmentParam::__FIELDS__();
 }
 
-#if MXNET_USE_OPENCV
 
 #ifdef _MSC_VER
 #define M_PI CV_PI

--- a/src/io/image_det_aug_default.cc
+++ b/src/io/image_det_aug_default.cc
@@ -15,10 +15,7 @@
 namespace mxnet {
 namespace io {
 
-#if MXNET_USE_OPENCV
-
 using nnvm::Tuple;
-using Rect = cv::Rect_<float>;
 
 namespace image_det_aug_default_enum {
 enum ImageDetAugDefaultCropEmitMode {kCenter, kOverlap};
@@ -186,6 +183,9 @@ std::vector<dmlc::ParamFieldInfo> ListDefaultDetAugParams() {
   return DefaultImageDetAugmentParam::__FIELDS__();
 }
 
+  
+#if MXNET_USE_OPENCV
+using Rect = cv::Rect_<float>;
 
 #ifdef _MSC_VER
 #define M_PI CV_PI

--- a/src/io/iter_image_recordio_2.cc
+++ b/src/io/iter_image_recordio_2.cc
@@ -19,6 +19,7 @@
 #include "./image_augmenter.h"
 #include "./image_iter_common.h"
 #include "./inst_vector.h"
+#include "../common/utils.h"
 
 namespace mxnet {
 namespace io {

